### DISCUSSION TITLE
Build errors are not actionable

### DISCUSSION THREAD

**Discussion by fkuehnel** [Homebrew/discussions#3661] [Everyday usage]
Maybe this is a repeated topic, however, this is an important one for a good build system:
Most, if not all of the errors reported from brew and cmake are not actionable, example a failed build on my side:

```
Last 15 lines from /Users/frank/Library/Logs/Homebrew/opencv/02.cmake:
cd /tmp/opencv-20220916-98858-g220qq/opencv-4.6.0/build_shared/modules/calib3d && /Users/frank/homebrew/Library/Homebrew/shims/mac/super/clang++ -DCVAPI_EXPORTS -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/opencv-20220916-98858-g220qq/opencv-4.6.0/build_shared -I/tmp/opencv-20220916-98858-g220qq/opencv-4.6.0/modules/calib3d/include -I/tmp/opencv-20220916-98858-g220qq/opencv-4.6.0/build_shared/modules/calib3d -I/tmp/opencv-20220916-98858-g220qq/opencv-4.6.0/modules/core/include -I/tmp/opencv-20220916-98858-g220qq/opencv-4.6.0/modules/flann/include -I/tmp/opencv-20220916-98858-g220qq/opencv-4.6.0/modules/imgproc/include -I/tmp/opencv-20220916-98858-g220qq/opencv-4.6.0/modules/features2d/include -isystem /Users/frank/homebrew/include/eigen3 -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-poi
```
Lot's more gibberish

What has caused it, and what is the course of action to take... no good hints besides deep digging of your own!

For once, I would like us to get to a good build system with good brief diagnostics and possible actions to take. Thank you! 

P.S.
Was trying to install OpenCV on my M1 mac!

**Comment by gromgit** [comment]
> What has caused it, and what is the course of action to take... no good hints besides deep digging of your own!

As the "gibberish" you posted begins:
>```
>Last 15 lines from /Users/frank/Library/Logs/Homebrew/opencv/02.cmake:
>```

That file contains the full build log, including the error(s) that cause the build to fail. Because the build process runs in parallel by default, these errors may actually be quite far from the end of the file.

>For once, I would like us to get to a good build system with good brief diagnostics and possible actions to take. Thank you!

From this and you "gibberish" comment, I assume you're not a C++ developer (and perhaps not a developer at all). This is par for the course when building from source on end-user platforms with some unknown set of extras and missing bits, which is why Homebrew provides pre-built bottles that would've been used in normal Homebrew setups, and why "build from source on end-user platforms" is _not_ supported by the Homebrew crew.

What's the output of `brew config` and `brew doctor` on your Mac?

**Comment by SMillerDev** [comment]
You're in the wrong place for this. Homebrew is not a build system, it only invokes the cmake build system in this case.
